File: samples/client/3_1_0_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/IfAppearsAtTheEndWhenSerializedKeywordProcessingSequenceTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.MapUtils;
import org.checkerframework.checker.nullness.qual.Nullable;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.AbstractMap;

public class IfAppearsAtTheEndWhenSerializedKeywordProcessingSequenceTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.onlyFormat());

    @Test
    public void testInvalidRedirectsToElseAndFailsFails() {
        // invalid redirects to else and fails
        final var schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance();
        try {
            schema.validate(
                "invalid",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }

    @Test
    public void testYesRedirectsToThenAndPassesPasses() throws ValidationException {
        // yes redirects to then and passes
        final var schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance();
        schema.validate(
            "yes",
            configuration
        );
    }

    @Test
    public void testOtherRedirectsToElseAndPassesPasses() throws ValidationException {
        // other redirects to else and passes
        final var schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance();
        schema.validate(
            "other",
            configuration
        );
    }

    @Test
    public void testNoRedirectsToThenAndFailsFails() {
        // no redirects to then and fails
        final var schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance();
        try {
            schema.validate(
                "no",
                configuration
            );
            throw new RuntimeException("A different exception must be thrown");
        } catch (ValidationException ignored) {
            ;
        }
    }
}

File: samples/client/3_1_0_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/IfAppearsAtTheEndWhenSerializedKeywordProcessingSequenceTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class IfAppearsAtTheEndWhenSerializedKeywordProcessingSequenceTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testInvalidRedirectsToElseAndFailsFails() {
        // invalid redirects to else and fails
        val schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "invalid",
                    configuration
                )
            }
        )
    }

    @Test
    fun testYesRedirectsToThenAndPassesPasses() {
        // yes redirects to then and passes
        val schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance()
        schema.validate(
            "yes",
            configuration
        )
    }

    @Test
    fun testOtherRedirectsToElseAndPassesPasses() {
        // other redirects to else and passes
        val schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance()
        schema.validate(
            "other",
            configuration
        )
    }

    @Test
    fun testNoRedirectsToThenAndFailsFails() {
        // no redirects to then and fails
        val schema = IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence.IfAppearsAtTheEndWhenSerializedKeywordProcessingSequence1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    "no",
                    configuration
                )
            }
        )
    }
}
